File: app/src/main/java/tech/thdev/app/view/recycler/adapter/RecyclerBottomSheetAdapter.java
package tech.thdev.app.view.recycler.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import java.util.ArrayList;

import tech.thdev.app.data.BottomData;

/**
 * Created by Tae-hwan on 09/12/2016.
 */

public class RecyclerBottomSheetAdapter extends RecyclerView.Adapter<RecyclerBottomSheetViewHolder> {

    private ArrayList<BottomData> itemList = new ArrayList<>();

    private Context context;

    public RecyclerBottomSheetAdapter(Context context) {
        this.context = context;
    }

    public void setItem(BottomData item) {
        itemList.add(item);
    }

    public BottomData getItem(int position) {
        return itemList.get(position);
    }

    @Override
    public RecyclerBottomSheetViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new RecyclerBottomSheetViewHolder(context, parent);
    }

    @Override
    public void onBindViewHolder(RecyclerBottomSheetViewHolder holder, int position) {
        holder.onBind(getItem(position), position);
    }

    @Override
    public int getItemCount() {
        return itemList.size();
    }
}

File: app/src/main/java/tech/thdev/app/view/recycler/adapter/RecyclerBottomSheetAdapter.kt
package tech.thdev.app.view.recycler.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import tech.thdev.app.data.BottomData
import java.util.*

/**
 * Created by Tae-hwan on 09/12/2016.
 */
class RecyclerBottomSheetAdapter : RecyclerView.Adapter<RecyclerBottomSheetViewHolder>() {

    private val itemList = ArrayList<BottomData>()
    fun setItem(item: BottomData) {
        itemList.add(item)
    }

    fun getItem(position: Int): BottomData {
        return itemList[position]
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): RecyclerBottomSheetViewHolder {
        return RecyclerBottomSheetViewHolder(parent)
    }

    override fun onBindViewHolder(holder: RecyclerBottomSheetViewHolder, position: Int) {
        holder.onBind(getItem(position))
    }

    override fun getItemCount(): Int {
        return itemList.size
    }
}